package.json

```diff
@@ -16,6 +16,7 @@
         "default": "./dist/index.esm.mjs"
       },
       "types": "./dist/index.d.ts",
+      "module": "./dist/index.esm.js",
```

Comment:
AFAR, vite will pick "import" condition before "module" condition, no?

Reply:
That's a good question! I have yet to test this with Next Js and Vite 😅. I was thinking to have `.mjs` picked up for Vite since it has better `import.meta.*` support. 

I'm trying to get it to work with CRA. It appears that Webpack needs a `index.ts` file with a _single_ condition export like [this](https://github.com/jotaijs/jotai-devtools/blob/fix/export-esm-with-process-env/src/DevTools/DevTools.tsx), otherwise, it does not seem to be "tree-shaking" out. However, TSUP chunks everything into one file, causing Webpack to bundle this in.

Reply:
FYI, vite will replace `import.meta.env.MODE`, but **not** `import.meta.env`, so it doesn't eliminate DEV code unfortunately. I didn't test it, it's just an assumption from the docs.

Reply:
> It's something we dropped in jotai v2. [pmndrs/jotai#1675](https://github.com/pmndrs/jotai/pull/1675) If this solves some cases, should we revert it? cc: @barelyhuman

If the devtools are just supporting v2 then I think the webpack 4 support is dropped anyway. 
Though it's not a big deal to revert it so can be done if it helps the devtools

As for Vite
It will pickup exports instead of the `module` key. It's based off lukeed/resolve.exports

https://github.com/lukeed/resolve.exports/blob/ac545e5b1b0845c0cae9d5e728531db85becc3fa/src/index.ts#L7

I've not gone through the webpack 5 implementation so can't say about it

Reply:
The export conditions did not work as expected, so I dropped the support for `import.meta.env` in favor of `process.env.NODE_ENV` and it works as expected across all 3 bundlers (CRA, NextJS, and Vite). 

Any thoughts? Should we go forward with it?

Reply:
We can go with it for jotai-devtools.

I would like to revisit zustand/jotai/valtio based on your expertise.

> The export conditions did not work as expected

That's bad news though.

Reply:
For sure! Happy to help!

> That's bad news though.

Agreed! Other libraries (like Redux) are taking sticking to `process.env.*`, too. Would be good to revisit it once Webpack and some other popular bundlers add it.